packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx

```diff
@@ -2,18 +2,45 @@ import React, { useCallback, useState } from 'react';
 import classNames from 'classnames';
 import { launchPatientWorkspace, useOrderBasket } from '@openmrs/esm-patient-common-lib';
 import { useLayoutType, useSession } from '@openmrs/esm-framework';
-import { type LabOrderBasketItem, careSettingUuid, prepLabOrderPostData } from '../api';
-import { Button, ButtonSet, Column, ComboBox, Form, Layer, Grid, InlineNotification, TextInput } from '@carbon/react';
+import { careSettingUuid, type LabOrderBasketItem, prepLabOrderPostData } from '../api';
+import {
+  Button,
+  ButtonSet,
+  Column,
+  ComboBox,
+  Form,
+  Layer,
+  Grid,
+  InlineNotification,
+  TextInput,
+  FormLabel,
+} from '@carbon/react';
 import { useTranslation } from 'react-i18next';
 import { priorityOptions } from './lab-order';
-import { type TestType, useTestTypes } from './useTestTypes';
+import { useTestTypes } from './useTestTypes';
 import styles from './lab-order-form.scss';
+import { Controller, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
 
 export interface LabOrderFormProps {
   initialOrder: LabOrderBasketItem;
   closeWorkspace: () => void;
 }
 
+const labOrderFormSchema = z.object({
+  urgency: z.string({ required_error: 'Priority is required', invalid_type_error: 'Priority is required' }),
+  instructions: z.string().optional(),
+  labReferenceNumber: z.string({
+    required_error: 'Lab reference number is required',
+    invalid_type_error: 'Lab reference number is required',
+  }),
+  testType: z.object(
+    { label: z.string(), conceptUuid: z.string() },
+    { required_error: 'Test type is required', invalid_type_error: 'Test type is required' },
```

Comment:
AFAICT, [invalid_type_error](https://zod.dev/?id=strings) is used to specify a custom error message that will be displayed when the type of a value does not match the expected type defined in the schema. So, in this case, it will complain if the values provided in the validated fields are not strings. The error message provided should be reflective of that. Or if we're not validating type mismatches, we could remove it entirely.

Reply:
Hi, @denniskigen!
We should use `invalid_type_error` for objects, because the error thrown by Zod in case the input is removed says "Expected object, recieved null".

@usamaidrsk, we don't need need `invalid_type_error` messages for string, required_error should handle that for the strings. Please see if removing `invalid_type_error` for z.object doesn't throw the error I shared above, i.e. `required_error` takes care of it or not.
Thanks!

Reply:
Thanks @denniskigen 
Here I used the same error message as of required, just because `Expected string, received null` would not give clear meaning to the user on the form.

If there is a suggestion, I would like to know, then I can replace with that.